app/src/main/java/net/kwatts/powtools/drawables/TrackDrawable.kt

```diff
@@ -0,0 +1,68 @@
+package net.kwatts.powtools.drawables
+
+import android.graphics.*
+import android.graphics.drawable.Drawable
+import kotlin.math.absoluteValue
+
+class TrackDrawable @JvmOverloads constructor(
+        track: List<Pair<Double?, Double?>>,
+        private val scale: Float = 1f,
+        private val trackWidth: Float = 1f,
+        private val trackColor: Int = Color.BLACK
+) : Drawable() {
+
+    private val paint: Paint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        color = trackColor
+        strokeWidth = trackWidth
+    }
+
+    private val track: List<Pair<Double, Double>> = track
+            .filter { (latitude, longitude) -> latitude != null && longitude != null }
+            .map { Pair(it.first!! * scale, it.second!! * scale) }
```

Comment:
maybe just mention that scale is used to prevent loss of precision because deltas in gps are so small?

Reply:
Added KDoc to `TrackDrawable` class.